bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BFloat.java

```diff
@@ -98,7 +99,12 @@ public Double value() {
     }
 
     @Override
-    public BValue copy() {
-        return new BFloat(value);
+    public BValue copy(Map<BValue, BValue> refs) {
+        if (refs.containsKey(this)) {
```

Comment:
even this should be return this; check other value types as well

Reply:
fixed